packages/SwingSet/src/vats/vattp/vat-vattp.js

```diff
@@ -163,24 +171,27 @@ export function buildRootObject(vatPowers, _vatParams, baggage) {
   //   E(agoric.ibcport[1]).connect('/ibc-port/portADDR/ordered/vattp-1', handler);
   //   E(E(host).lookup(helloAddress)).hello();
 
-  const nextConnectionNumberBaggageKey = 'networkHostNextConnectionNumber';
-  let nextConnectionNumber = provide(
+  // Define the durable NetworkHost kind
+  // and retain a durable monotonic connection number counter and set of names.
+  // TODO: Should this collection be Weak (non-iterable)?
+  const networkHostNames = provideDurableSetStore(
     baggage,
-    nextConnectionNumberBaggageKey,
-    () => 7n,
+    networkHostNamesBaggageKey,
+  );
+  let networkHostCounter = provide(
+    baggage,
+    networkHostCounterBaggageKey,
+    () => 0n,
```

Comment:
that changes the initial value from 7n to 0n, but I'm cool with that, I don't think the non-zero starting point was buying us anything